modules/social_features/social_follow_content/social_follow_content.module

```diff
@@ -9,44 +9,65 @@ use Drupal\Core\Entity\EntityInterface;
 use Drupal\flag\Entity\Flagging;
 
 /**
- * Implements hook_ENTIT_YTYPE_insert().
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function social_follow_content_comment_insert(EntityInterface $entity) {
+  if ($entity->getCommentedEntity()->getEntityTypeId() == 'post') {
+    _social_follow_content_action($entity);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
  */
 function social_follow_content_event_enrollment_insert(EntityInterface $entity) {
-  social_follow_content_event_enrollment_follow($entity);
+  _social_follow_content_action($entity);
 }
 
 /**
- * Implements hook_ENTIT_YTYPE_update().
+ * Implements hook_ENTITY_TYPE_update().
  */
 function social_follow_content_event_enrollment_update(EntityInterface $entity) {
-  social_follow_content_event_enrollment_follow($entity);
+  _social_follow_content_action($entity);
 }
 
 /**
- * Automatically follow content when user enrolls the event.
+ * Automatically follow content or post.
  */
-function social_follow_content_event_enrollment_follow(EntityInterface $entity) {
-  if ($entity->field_enrollment_status->value) {
-    // Check if user already follow this content.
-    $account = \Drupal::currentUser();
+function _social_follow_content_action(EntityInterface $entity) {
```

Comment:
This function has so many checks for `if ($event)` in it that it is probably better to just create a separate function to handle when a user comments on a post. Any code that is then the same between this function and the new function could be extracted for re-use to a separate function to keep things DRY.